build.gradle

```diff
@@ -41,7 +41,7 @@ plugins {
 
   id "com.github.spotbugs" version '6.0.25' apply false
   id 'org.scoverage' version '8.0.3' apply false
-  id 'io.github.goooler.shadow' version '8.1.3' apply false
+  id 'io.github.goooler.shadow' version '8.1.8' apply false
```

Comment:
Could you try the successor (https://github.com/GradleUp/shadow)? It may fix the issue like https://issues.apache.org/jira/browse/KAFKA-17052 and https://issues.apache.org/jira/browse/KAFKA-16359

Reply:
@chia7712 Yeah, I tried to upgrade Goooler/shadow to GradleUp/shadow, and after the upgrade, the KAFKA-17052 problem did not occur.But as @chia7712  mentioned, This is because the GradleUp/shadow 8.3.0 release does not have the Goooler/shadow#80 patch that caused us to revert versions.Also, unfortunately, GradleUp/shadow does not fix the issue that caused KAFKA-16359.However, since Goooler/shadow is retired now.so I will create a new ticket for changing the fork.